diff --git a/Platform/RaspberryPi/Library/PlatformBootManagerLib/PlatformBm.c b/Platform/RaspberryPi/Library/PlatformBootManagerLib/PlatformBm.c
index ab8fdb586f..0e2c532dd5 100644
--- a/Platform/RaspberryPi/Library/PlatformBootManagerLib/PlatformBm.c
+++ b/Platform/RaspberryPi/Library/PlatformBootManagerLib/PlatformBm.c
@@ -367,6 +367,12 @@ PlatformRegisterBootOption (
            );
   ASSERT_EFI_ERROR (Status);
 
+  //
+  // Connect all devices, and regenerate all boot options
+  //
+  EfiBootManagerConnectAll ();
+  EfiBootManagerRefreshAllBootOption ();
+
   BootOptions = EfiBootManagerGetLoadOptions (&BootOptionCount, LoadOptionTypeBoot);
 
   OptionIndex = EfiBootManagerFindLoadOption (&NewOption, BootOptions, BootOptionCount);
@@ -818,7 +824,19 @@ PlatformBootManagerUnableToBoot (
   //
   BootOptions = EfiBootManagerGetLoadOptions (&NewBootOptionCount,
                   LoadOptionTypeBoot);
+
+  // Always try PXE after boot failure
+  for (Index=0; Index < NewBootOptionCount; Index++) {
+    if (BootOptions[Index].FilePath->Type == MESSAGING_DEVICE_PATH &&
+      BootOptions[Index].FilePath->SubType == MSG_MAC_ADDR_DP) {
+      DEBUG ((DEBUG_WARN, "%a: trying to boot option %i\n",
+        __FUNCTION__, Index));
+      EfiBootManagerBoot (&BootOptions[Index]);
+      break;
+    }
+  }
   EfiBootManagerFreeLoadOptions (BootOptions, NewBootOptionCount);
+  return;
 
   //
   // If the number of configured boot options has changed, reboot
